core/trino-spi/src/main/java/io/trino/spi/connector/ConnectorMetadata.java

```diff
@@ -779,6 +780,25 @@ default Optional<TrinoPrincipal> getSchemaOwner(ConnectorSession session, Catalo
         return Optional.empty();
     }
 
+    /**
+     * Attempt to push down an update operation into the connector. If a connector
+     * can execute an update for the table handle on its own, it should return a
+     * table handle, which will be passed back to {@link #executeUpdate} during
+     * query executing to actually execute the update.
+     */
+    default Optional<ConnectorTableHandle> applyUpdate(ConnectorSession session, ConnectorTableHandle handle)
+    {
+        return Optional.empty();
+    }
+
+    /**
+     * Execute the update operation on the handle returned from {@link #applyUpdate}.
+     */
+    default OptionalLong executeUpdate(ConnectorSession session, ConnectorTableHandle handle, Map<String, Domain> assignments)
+    {
+        throw new TrinoException(GENERIC_INTERNAL_ERROR, "ConnectorMetadata applyUpdate() is implemented without executeDelete()");
```

Comment:
` "ConnectorMetadata applyUpdate() is implemented without executeUpdate()"`

Reply:
fixed